redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsTerminationService.java

```diff
@@ -61,7 +66,9 @@ private DatabaseServerConfig terminate(DatabaseServerConfig server, boolean forc
         if (LOGGER.isDebugEnabled()) {
             LOGGER.debug("Terminate called for: {} with force: {}", dbStack, force);
         }
-        if (dbStack.getStatus().isDeleteInProgressOrCompleted() && !deletetedOnProviderSide(server, dbStack)) {
+
+        if (flowLogService.isFlowConfigAlreadyRun(dbStack.getId(), RedbeamsTerminationFlowConfig.class)
+                || (server.isArchived() && dbStack.getStatus().isDeleteCompleted())) {
```

Comment:
why have you modified the second clause, the private method was more descriptive, what's the goal of removing it?

Reply:
the logic is changed there and the name isn't matched for the logic. Here we should skip the deletion if archived and delete completed. The method was this: it is not archived and deleted. However, I can change the method name to already deleted or smth like this